### ISSUE TITLE
Big db.collection.deleteMany()/Collection.remove() ops cause mongo/oplog to crash?

### ISSUE THREAD

**Issue by leosco** [meteor/meteor#8405]
When deleting large amounts of bulk data ~20000 documents/50 fields using `Collection.remove()` or `Collection.rawCollection().deleteMany()` the node process crashes with a RangeError, after the db operation completes. I'm having this problem currently on `METEOR@1.4.2.7`
```
Exception in defer callback: RangeError: Out of memory
     at RangeError (native)
     at DocFetcher.fetch (packages/mongo/doc_fetcher.js:60:8)
     at packages/mongo/oplog_observe_driver.js:509:43
     at [object Object]._.extend.forEach (packages\id-map.js:68:35)
     at packages/mongo/oplog_observe_driver.js:507:35
     at packages/mongo/oplog_observe_driver.js:16:9
     at [object Object]._.extend.withValue (packages\meteor.js:1122:17)
     at packages\meteor.js:445:45
     at runWithEnvironment (packages\meteor.js:1176:24)
 (STDERR) FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory
```
After chatting about it on the gitter with @copleykj I confirmed that the same operation worked without crashing after I disabled websockets and ran the method over long-polling. This could suggest that `packages/mongo/oplog_observe_driver.js` is running out of memory trying to update the oplog. I'm not sure if error occurs during other db operations, but I've only encountered it with `Collection.remove()`/`deleteMany()`. 

Update: I've also encountered this when performing this operation directly from the mongo shell, while running my app in development using `db.collection.remove()`. A couple of seconds after I get
```
WriteResult({ "nRemoved" : 20000 })
```
in the mongo shell, I then get in the server logs
```
(STDERR) FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory
```

However, it doesn't occur as consistently as when the call to the db is made in my server-side code.

**Comment by leosco** [comment]
Update: I've also encountered this when performing this operation directly from the mongo shell, while running my app in development using `db.collection.remove()`. A couple of seconds after I get
```
WriteResult({ "nRemoved" : 20000 })
```
in the mongo shell, I then get in the server logs
```
(STDERR) FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory
```

However, it doesn't occur as consistently as when the call to the db is made in my server-side code.

**Comment by abernix** [comment]
Interesting!  Do you mind making an easy, slim reproduction that recreates this?  Just something that puts in the rows and deletes them (or what have you)?

**Comment by leosco** [comment]
@abernix I've been trying to reproduce the issue on a fresh project but, so far, it's only occurring on a large project that's been in development for over a year. As you can imagine, it has all sorts of dependencies and different things going on. It also has been migrated from many previous Meteor releases and node versions, so if there's some freak edge case being tripped along the way it could be very difficult for me to reproduce it. 

If time permits, I'll start adding in dependencies such as collection2 one by one to try and recreate the original error environment and get back to you. I'm also doing all my development on Windows 10, just FYI. 

Is it at all possible that this could be related to some of the other phantom errors that people are encountering later down the road with large Meteor projects, such as https://github.com/meteor/meteor/issues/8002 or https://github.com/meteor/meteor/issues/8057?

**Comment by leosco** [comment]
This....may have been resolved in 1.6? My tests for the affected module seem to be doing alright now.

I'm just going to close this for now and I'll only open it again if I can reproduce it properly.
